### PR TITLE
Updating datastore system test to allow concurrent runs.

### PR DESCRIPTION
**Pull request by dhermes**
Using a namespace for any key which is changed (inserted/deleted) so that concurrent tests don't operate on the same data. The one exception is `TestDatastoreQuery` which depends on entities which need to be already stored and indexed before running. For this test case, a new client is created without any namespace.

**NOTE**: Has #1369 as diffbase.

Inspired by #1363.

### CODE REVIEW

File: system_tests/datastore.py [hunk 1]
```diff
@@ -41,18 +44,22 @@ class Config(object):
...
     else:
-        Config.CLIENT = datastore.Client(dataset_id=EMULATOR_DATASET)
-        populate_datastore.add_characters(client=Config.CLIENT)
-
-
-def tearDownModule():
-    if EMULATOR_DATASET is not None:
-        clear_datastore.remove_all_entities(client=Config.CLIENT)
+        Config.CLIENT = datastore.Client(dataset_id=EMULATOR_DATASET,
+                                         namespace=TEST_NAMESPACE)
```
Comment (tseaver) [inline system_tests/datastore.py]: Why are we setting the namespace only when `EMULATOR_DATASET` is present?  Wouldn't it be better to use it unconditionally?

Comment (dhermes) [inline system_tests/datastore.py]: Check again. The `namespace **is** used always.

**Comment by dhermes** [comment]
@tseaver FYI #1383 should be merged before this.

**Comment by dhermes** [comment]
LuLz. Pushed a test key.

File: system_tests/datastore.py [hunk 2]
```diff
@@ -41,18 +44,22 @@ class Config(object):
...
     else:
-        Config.CLIENT = datastore.Client(project=EMULATOR_DATASET)
-        populate_datastore.add_characters(client=Config.CLIENT)
-
-
-def tearDownModule():
-    if EMULATOR_DATASET is not None:
-        clear_datastore.remove_all_entities(client=Config.CLIENT)
+        Config.CLIENT = datastore.Client(project=EMULATOR_DATASET,
+                                         namespace=TEST_NAMESPACE)
```
Comment (dhermes) [inline system_tests/datastore.py]: @tseaver PTAL. The issue was that `namespace` wasn't in both cases but it is, so this should be all good?

Comment (tseaver) [inline system_tests/datastore.py]: I can see that now.  I guess I mis-read the diff:  the lack of parallelism between the two clauses might be what threw me.

LGTM

**Pull request merged**